.github/prFormat.js

```diff
@@ -0,0 +1,43 @@
+const fs = require('fs');
+
+const readMaintainers = async (path) => {
+  const rawMaintainers = await fs.promises.readFile(path, {encoding: 'utf8'});
+  return rawMaintainers.trim().split('\n')
+    .map((maintainer) => `@${maintainer}`)
+    .join(' ');
+};
+
+
+module.exports = async ({ github, context: { payload }, maintainersFile }) => {
+  // https://developer.github.com/v3/search/#search-issues-and-pull-requests
+ const { data: { total_count, items: [pr] } } = await github.search.issuesAndPullRequests({
+    q: `${payload.head_commit.id}+repo:${payload.repository.full_name}+type:pr+is:merged`,
+  });
+
+  // Use pr title if available or fallback to first line of head commit message
+  const title = total_count
+    ? pr.title
+    : payload.head_commit.message.split('\n')[0];
```

Comment:
Given this is being triggered by pushes to `master` (not pull requests), won't it never have a PR object?  (and in that case, the `head_commit` is going to be the merge commit, and thus generate a problematic title/commit message)

Reply:
I do a search for PRs with the sha of the head_commit that was pushed.  It's a similar approach to how [semantic-release](https://github.com/semantic-release/semantic-release/blob/master/index.js) looks for issues/PRs